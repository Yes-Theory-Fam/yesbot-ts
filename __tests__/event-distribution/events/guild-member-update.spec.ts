import {
  addGuildMemberUpdateHandler,
  GuildMemberUpdateEventHandlerOptions,
} from "../../../src/event-distribution/events/guild-member-update.js";
import MockDiscord from "../../mocks/index.js";
import {
  InstanceOrConstructor,
  StringIndexedHIOCTree,
} from "../../../src/event-distribution/types/hioc.js";
import {
  CommandHandler,
  DiscordEvent,
} from "../../../src/event-distribution/index.js";

describe("GuildMemberUpdates", () => {
  let mockDiscord: MockDiscord;
  beforeEach(() => {
    mockDiscord = new MockDiscord();
  });

  // it("should extract keys despite no role differences", () => {
  //   const oldMember = mockDiscord.getGuildMember();
  //   const newMember = mockDiscord.getGuildMember();
  //   newMember.nickname = "Test";
  //   const result = extractGuildMemberUpdateInfo(oldMember, newMember);
  //   expect(result).toMatchSnapshot();
  // });
  //
  // it("should extract keys with a single role difference", () => {
  //   const oldMember = mockDiscord.getGuildMember();
  //   const newMember = mockDiscord.getGuildMember();
  //   const role = mockDiscord.getRole();
  //   newMember.roles.cache.set(role.id, role);
  //   const result = extractGuildMemberUpdateInfo(oldMember, newMember);
  //   expect(result).toMatchSnapshot();
  // });
  //
  // it("should extract keys with multiple role differences", () => {
  //   const oldMember = mockDiscord.getGuildMember();
  //   const newMember = mockDiscord.getGuildMember();
  //   const role = mockDiscord.getRole();
  //   const role2 = mockDiscord.getRole();
  //   newMember.roles.cache.set(role.id, role);
  //   newMember.roles.cache.set(role2.id, role2);
  //   const result = extractGuildMemberUpdateInfo(oldMember, newMember);
  //   expect(result).toMatchSnapshot();
  // });

  it("should add data to the tree", () => {
    const tree: StringIndexedHIOCTree<DiscordEvent.GUILD_MEMBER_UPDATE> = {};
    const options: GuildMemberUpdateEventHandlerOptions = {
      event: DiscordEvent.GUILD_MEMBER_UPDATE,
      roleNamesAdded: ["Test"],
      roleNamesRemoved: ["Example"],
    };
    const ioc = {} as InstanceOrConstructor<
      CommandHandler<DiscordEvent.GUILD_MEMBER_UPDATE>
    >;

    addGuildMemberUpdateHandler(options, ioc, tree);
    expect(tree).toMatchSnapshot();

    const secondOptions: GuildMemberUpdateEventHandlerOptions = {
      event: DiscordEvent.GUILD_MEMBER_UPDATE,
    };
    addGuildMemberUpdateHandler(secondOptions, ioc, tree);
    expect(tree).toMatchSnapshot();

    const thirdOptions: GuildMemberUpdateEventHandlerOptions = {
      event: DiscordEvent.GUILD_MEMBER_UPDATE,
      roleNamesAdded: ["Test"],
    };
    addGuildMemberUpdateHandler(thirdOptions, ioc, tree);
    expect(tree).toMatchSnapshot();
  });
});
