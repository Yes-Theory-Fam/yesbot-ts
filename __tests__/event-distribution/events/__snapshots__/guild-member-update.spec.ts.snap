// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuildMemberUpdates should add data to the tree 1`] = `
{
  "added": {
    "Test": [
      {
        "ioc": {},
        "options": {
          "event": "GUILD_MEMBER_UPDATE",
          "roleNamesAdded": [
            "Test",
          ],
          "roleNamesRemoved": [
            "Example",
          ],
        },
      },
    ],
  },
  "removed": {
    "Example": [
      {
        "ioc": {},
        "options": {
          "event": "GUILD_MEMBER_UPDATE",
          "roleNamesAdded": [
            "Test",
          ],
          "roleNamesRemoved": [
            "Example",
          ],
        },
      },
    ],
  },
}
`;

exports[`GuildMemberUpdates should add data to the tree 2`] = `
{
  "": {
    "": [
      {
        "ioc": {},
        "options": {
          "event": "GUILD_MEMBER_UPDATE",
        },
      },
    ],
  },
  "added": {
    "Test": [
      {
        "ioc": {},
        "options": {
          "event": "GUILD_MEMBER_UPDATE",
          "roleNamesAdded": [
            "Test",
          ],
          "roleNamesRemoved": [
            "Example",
          ],
        },
      },
    ],
  },
  "removed": {
    "Example": [
      {
        "ioc": {},
        "options": {
          "event": "GUILD_MEMBER_UPDATE",
          "roleNamesAdded": [
            "Test",
          ],
          "roleNamesRemoved": [
            "Example",
          ],
        },
      },
    ],
  },
}
`;

exports[`GuildMemberUpdates should add data to the tree 3`] = `
{
  "": {
    "": [
      {
        "ioc": {},
        "options": {
          "event": "GUILD_MEMBER_UPDATE",
        },
      },
    ],
  },
  "added": {
    "Test": [
      {
        "ioc": {},
        "options": {
          "event": "GUILD_MEMBER_UPDATE",
          "roleNamesAdded": [
            "Test",
          ],
          "roleNamesRemoved": [
            "Example",
          ],
        },
      },
      {
        "ioc": {},
        "options": {
          "event": "GUILD_MEMBER_UPDATE",
          "roleNamesAdded": [
            "Test",
          ],
        },
      },
    ],
  },
  "removed": {
    "Example": [
      {
        "ioc": {},
        "options": {
          "event": "GUILD_MEMBER_UPDATE",
          "roleNamesAdded": [
            "Test",
          ],
          "roleNamesRemoved": [
            "Example",
          ],
        },
      },
    ],
  },
}
`;
